lib/core/asynctree.js

```diff
@@ -88,7 +88,7 @@ class AsyncTree extends EventEmitter{
   }
 
   shouldRejectNodePromise(err) {
-    if ((err.isExpect|| this.currentNode.namespace === 'assert') && this.currentNode.isES6Async) {
+    if ((err.isExpect|| this.currentNode.namespace === 'assert' || err.abortOnFailure) && this.currentNode.isES6Async) {
```

Comment:
abortOnFailure is probably not the right flag. For async functions should we not reject the promise in all cases? 
@beatfactor @gravityvi why are we rejecting `isExpect` and `assert` selectively?

Reply:
Just adding this line change to my project can fix the issue. But I need to find out the side effects mentioned above. @gravityvi  can you give me some details about what unhandled rejection will be caused in what kind of commands? Some example?